### PR TITLE
VIRTS 3461 - Browse Youtube

### PR DESCRIPTION
**Pull request by kaylakraines**
## Description

Add a new behavior to the human plugin to open a Chrome Browser, browse to youtube.com, perform a search on youtube, select results, pause to watch videos, and click on suggested videos. Random sleep times are incorporated between each action to simulate human browsing activity.

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by deacon-mp** [comment]
> ## Description
> Add a new behavior to the human plugin to open a Chrome Browser, browse to youtube.com, perform a search on youtube, select results, pause to watch videos, and click on suggested videos. Random sleep times are incorporated between each action to simulate human browsing activity.
> 
> ## Type of change
> Please delete options that are not relevant.
> 
> * [ ]  New feature (non-breaking change which adds functionality)
> * [ ]  This change requires a documentation update
> 
> ## How Has This Been Tested?
> Please describe the tests that you ran to verify your changes.
> 
> ## Checklist:
> * [ ]  My code follows the style guidelines of this project
> * [ ]  I have performed a self-review of my own code
> * [ ]  I have made corresponding changes to the documentation
> * [ ]  I have added tests that prove my fix is effective or that my feature works

Replying here

File: pyhuman/app/workflows/browse_youtube.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+MAX_SUGGESTED_VIDEOS = 10
+
+SEARCH_LIST = 'browse_youtube.txt'
+
+def load():
+    driver = WebDriverHelper()
+    return GoogleSearch(driver=driver)
+
+
+class GoogleSearch(BaseWorkflow):
```
Comment (djlawren) [inline pyhuman/app/workflows/browse_youtube.py]: I think this class name may be a holdover from a copy-paste. Probably a good idea to name it something like YoutubeBrowse or something like that.

File: pyhuman/app/workflows/browse_youtube.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        self._search_web()
+
+    """ PRIVATE """
+
+    def _search_web(self):
+        random_search = self._get_random_search()
+
+        # Navigate to youtube
+        self.driver.driver.get('https://www.youtube.com')
+        sleep(random.randrange(2, MAX_WAIT_TIME))
```
Comment (djlawren) [inline pyhuman/app/workflows/browse_youtube.py]: May also be a good idea to save the minimum wait time to a constant called MIN_WAIT_TIME.

**Comment by djlawren** [comment]
Tested and works. Just a couple really small comments. Looks good otherwise!

File: pyhuman/app/workflows/browse_youtube.py (hunk 1 above)
Comment (kaylakraines) [reply pyhuman/app/workflows/browse_youtube.py]: Oh gosh, thanks for catching that:)

**Review by djlawren — APPROVED** [review]
Looks good to me!

**Pull request merged**